src/ui/public/directives/input_number.js

```diff
@@ -0,0 +1,18 @@
+import uiModules from 'ui/modules';
+let module = uiModules.get('kibana');
+
+module.directive('inputNumber', function () {
+  return {
+    restrict: 'A',
+    require: 'ngModel',
+    link: function ($scope, $elem, attrs, ngModel) {
+      ngModel.$parsers.push(checkNumber);
```

Comment:
Just my preference but I would first define the function, then use it. Not sure if there is an official style for this kind of thing.  Up to you.